Training/Training/intlist.c

```diff
@@ -0,0 +1,285 @@
+
+	if (list->isDeleted == true) {
+		return E_LIST_DNE;
+	}
+
+	struct Node* newNode = CreateNode (data);
+
+	// If inserting at the beginning
+	if (ind == 0) {
+		newNode->next = list->head;
+		list->head = newNode;
+		return 0;
+	}
+
+	struct Node* current = list->head;
+	for (int i = 0; i < ind - 1 && current != NULL; ++i) {
+		current = current->next;
+	}
+
+	// If the position is out of bounds
+	if (current == NULL || ind < 0) {
+		free (newNode);
+		return E_INVALID_INDEX;
+	}
+
+	newNode->next = current->next;
+	current->next = newNode;
+	return 0;
+}
+
+
+/// <summary>
+/// The flag ensures whether the list exists.
+/// Removes the node at a specified index from the list.
+/// </summary>
+/// <param name="list"></param>
+/// <param name="ind"></param>
+/// <returns>Error code (if any)</returns>
+int RemoveAt (struct LinkedList* list, int ind) {
+
+	if (list->isDeleted == true) {
+		return E_LIST_DNE;
+	}
+
+	struct Node* temp = list->head;
+	struct Node* prev = NULL;
+
+	// Base case if linked list is empty
+	if (temp == NULL)
+		return E_EMPTY_LIST;
+
+	// Case 1: Head is to be deleted
+	if (ind == 0) {
+		list->head = temp->next;
+		free (temp);
+		return 0;
+	}
+
+	// Case 2: Node to be deleted is in middle
+	// Traverse till given position
+	for (int i = 0; temp != NULL && i < ind; i++) {
+		prev = temp;
+		temp = temp->next;
+	}
+
+	// If given index is found, delete node
+	if (temp != NULL && prev != NULL) {
+		prev->next = temp->next;
+		free (temp);
+		return 0;
+	} else {
+		return E_INVALID_INDEX;
+	}
+}
+
+
+/// <summary>
+/// The flag ensures whether the list exists.
+/// Removes the Node with the specified data value from the list.
+/// </summary>
+/// <param name="list"></param>
+/// <param name="data"></param>
+/// <returns>Error code (if any)</returns>
+int Remove (struct LinkedList* list, int data) {
+
+	if (list->isDeleted == true) {
+		return E_LIST_DNE;
+	}
+
+	// Store head node
+	struct Node* temp = list->head;
+	struct Node* prev = NULL;
+
+	if (temp == NULL)
+		return E_EMPTY_LIST;
+
+	// If head node itself holds the key to be deleted
+	if (temp != NULL && temp->data == data) {
+
+		// Changed head
+		list->head = temp->next;
+		free (temp);
+		return 0;
+	}
+
+	// Search for the key to be deleted, keep track of the
+	// previous node as we need to change 'prev->next'
+	while (temp != NULL && temp->data != data) {
+		prev = temp;
+		temp = temp->next;
+	}
+
+	// If key was not present in linked list
+	if (temp != NULL) {
+		prev->next = temp->next;
+		free (temp);
+		return 0;
+	} else
+		return E_INVALID_ELEMENT;
+
+}
+
+
+/// <summary>
+/// The flag ensures whether the list exists.
+/// Counts the number of nodes in the list.
+/// </summary>
+/// <param name="list"></param>
+/// <returns>count/Error code (if any)</returns>
+int Count (struct LinkedList* list) {
+
+	if (list->isDeleted == true) {
+		return E_LIST_DNE;
+	}
+
+	int count = 0;
+	while (list->head != NULL) {

```

Comment:
Since you are already keeping a variable to track list deletion, you can also keep a size/Count variable instead of traversing through the list again